src/response.rs

```diff
@@ -366,4 +405,17 @@ mod tests {
 
         assert_eq!(parsed_auth, expected_auth);
     }
+
+    #[test]
+    fn metrics_hierarchy_parent_of() {
+        let mut hierarchy = MetricsHierarchy::new();
+
+        hierarchy.insert(String::from("parent1"), vec![String::from("child1"),
+                                                       String::from("child2")]);
+        hierarchy.insert(String::from("parent2"), vec![String::from("child3")]);
+
+        assert_eq!(hierarchy.parent_of("child2"), Some("parent1"));
+        assert_eq!(hierarchy.parent_of("child3"), Some("parent2"));
+        assert_eq!(hierarchy.parent_of("nonchild"), None);
```

Comment:
Would be good to try with one of the parents for completeness.

Reply:
sounds good